### PR TITLE
Further cleanup of Condition access in AlCaDB, online DQM, Geometry, HLT, Simulation, and standard configuration; Further fixes for online DQM

### PR DESCRIPTION
**Pull request by diguida**
This is the next step of cleanup and improvement for condition access:
- Revision of the configuration initialisation and fragments for Condition 
  access:
  - Added `CondCore/ESSources/python/CondDBESSources_condDBv2_cfi.py` in order to keep compatibility with the explicit or implicit usage of the condition software version;
  - Added `Configuration/StandardSequences/python/AdditionalConditions_cff.py`: this fragment is used in order to load conditions not provided by Global Tag and/or Condition records;
  - Dropped `Configuration/StandardSequences/python/FrontierConditions_GlobalTag_cfi.py` and `Configuration/StandardSequences/python/FrontierConditions_GlobalTag_condDBv2_cfi.py`: they are not needed, as the initialisation of the `ESSource` for loading conditions is done in the proper package;
  - Dropped references to `Configuration/StandardSequences/python/FrontierConditions_GlobalTag_cfi.py`in client code.
  - Improved deprecation message for modules `CondCore/DBCommon/python/CondDBCommon_cfi.py` and `CondCore/DBCommon/python/CondDBSetup_cfi.py`.
- Define the default connection string after checking the cluster where the job is run: if in GPN or WAN, use the usual one, if in the online network, fetch conditions from the squid running on localhost.
- Revision of Condition access in Online DQM:
  - Cleanup of `DQM/Integration/python/config/FrontierCondition_GT_cfi.py` and `DQM/Integration/python/config/FrontierCondition_GT_autoExpress_cfi.py`: the connection string is established centrally;
  - Dropped `DQM/Integration/python/config/FrontierCondition_GT_Offline_cfi.py`;
  - Correspondent changes in client applications, where the GT customisation is enabled.
- Clean-up some `BuildFile` from dependencies upon removed packages.
- Improved test configuration file.
- Cleaned up spurious includes of `CondDBSetup_cfi` and `CondDBCommon_cfi` in `CondTools/DQM/python/DQMReferenceHistogramRootFileEventSetupAnalyzer_OrcoffOnly_cfi.py`, 
  `CondTools/DQM/python/DQMReferenceHistogramRootFileEventSetupAnalyzer_SQLiteOnly_cfi.py`, `CondTools/DQM/python/DQMReferenceHistogramRootFileEventSetupAnalyzer_cfi.py`, `Configuration/StandardSequences/python/RawToDigi_cff.py`, and `SimCalorimetry/HcalSimProducers/python/hcalUnsuppressedDigis_cfi.py`.

These are the fixes in order to run online DQM with 80X:
- Migration to consumes in `DPGAnalysis/Skims/interface/PhysDecl.h`, `DQM/TrigXMonitor/interface/L1Scalers.h`, `DQM/TrigXMonitor/interface/L1TScalersSCAL.h`, `EventFilter/L1GlobalTriggerRawToDigi/interface/L1GlobalTriggerRecordProducer.h`;
- Use latest HLT and express Global Tags;
- Load RECO geometry from GT in `DQM/Integration/python/clients/ecal_dqm_sourceclient-live_cfg.py`, `DQM/Integration/python/clients/ecalcalib_dqm_sourceclient-live_cfg.py`, `DQM/Integration/python/clients/hcal_dqm_sourceclient-live_cfg.py`
- Fixes in `DQM/Integration/python/clients/l1tstage2_dqm_sourceclient-live_cfg.py`
- Fixes in `DQM/DTMonitorModule/python/dt_dqm_sourceclient_common_cff.py` and `DQM/Integration/python/clients/dt_dqm_sourceclient-live_cfg.py` for loading correctly DT `TwinMux` sequence.

Originally in #13399 

Unit tests ran OK on `CMSSW_8_0_0`

```
$ git cms-ckeckdeps -a
$ scram b -r -j 4 runtests
```

Short matrix tests passed, and, as expected, no CondDB python `cfi` deprecation warning is seen on `CMSSW_8_0_0`

```
4.22_RunCosmics2011A+RunCosmics2011A+RECOCOSD+ALCACOSD+SKIMCOSD+HARVESTDC Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED Step4-PASSED  - time date Tue Feb  2 15:16:58 2016-date Tue Feb  2 15:12:56 2016; exit: 0 0 0 0 0
5.1_TTbar+TTbarFS+HARVESTFS Step0-PASSED Step1-PASSED  - time date Tue Feb  2 15:18:48 2016-date Tue Feb  2 15:12:57 2016; exit: 0 0
8.0_BeamHalo+BeamHaloINPUT+DIGICOS+RECOCOS+ALCABH+HARVESTCOS Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED Step4-PASSED  - time date Tue Feb  2 15:17:01 2016-date Tue Feb  2 15:13:00 2016; exit: 0 0 0 0 0
25.0_TTbar+TTbarINPUT+DIGI+RECOAlCaCalo+HARVEST+ALCATT Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED Step4-PASSED  - time date Tue Feb  2 15:20:38 2016-date Tue Feb  2 15:13:04 2016; exit: 0 0 0 0 0
134.911_RunSinglePh2015D+RunSinglePh2015D+HLTDR2_25ns+RECODR2_25nsreHLT+HARVESTDR2_25nsreHLT Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED  - time date Tue Feb  2 15:26:21 2016-date Tue Feb  2 15:16:59 2016; exit: 0 0 0 0
135.4_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS Step0-PASSED Step1-PASSED Step2-PASSED  - time date Tue Feb  2 15:29:02 2016-date Tue Feb  2 15:17:11 2016; exit: 0 0 0
140.53_RunHI2011+RunHI2011+RECOHID11+HARVESTDHI Step0-PASSED Step1-PASSED Step2-PASSED  - time date Tue Feb  2 15:22:55 2016-date Tue Feb  2 15:18:55 2016; exit: 0 0 0
1330.0_ZMM_13+ZMM_13INPUT+DIGIUP15+RECOUP15+HARVESTUP15 Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED  - time date Tue Feb  2 15:31:50 2016-date Tue Feb  2 15:20:46 2016; exit: 0 0 0 0
1000.0_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED Step4-PASSED  - time date Tue Feb  2 15:30:41 2016-date Tue Feb  2 15:22:57 2016; exit: 0 0 0 0 0
1001.0_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4 Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED Step4-PASSED Step5-PASSED Step6-PASSED  - time date Tue Feb  2 15:33:40 2016-date Tue Feb  2 15:26:31 2016; exit: 0 0 0 0 0 0 0
10 10 9 7 5 1 1 tests passed, 0 0 0 0 0 0 0 failed
```

Full matrix test running on `CMSSW_8_0_0`.

No regressions expected.

### CODE REVIEW

**Comment by diguida** [comment]
please test

**Comment by deguio** [comment]
please test

**Comment by Dr15Jones** [comment]
+1

**Comment by perrotta** [comment]
+1
(only a couple of old test configs affected for HLT)

**Comment by deguio** [comment]
+1

**Comment by civanch** [comment]
+1

**Comment by mmusich** [comment]
+1

**Comment by ggovi** [comment]
+1

**Comment by mmusich** [comment]
@diguida as foreseen this doesn't merge anymore after merging #13139. Can you please rebase?

**Comment by diguida** [comment]
please test

**Comment by diguida** [comment]
@mmusich rebase done. I also gave a look to the unit and short matrix test outputs, and, looking at the warning messages, I further cleaned up spurious includes in `CondTools/DQM/python/DQMReferenceHistogramRootFileEventSetupAnalyzer_OrcoffOnly_cfi.py`, 
`CondTools/DQM/python/DQMReferenceHistogramRootFileEventSetupAnalyzer_SQLiteOnly_cfi.py`, `CondTools/DQM/python/DQMReferenceHistogramRootFileEventSetupAnalyzer_cfi.py`, `Configuration/StandardSequences/python/RawToDigi_cff.py`, and `SimCalorimetry/HcalSimProducers/python/hcalUnsuppressedDigis_cfi.py`.
There is only one warning message issued in `1001.0`, but I want to see the new test output...

**Comment by diguida** [comment]
please test
oops one python file was mingled during rebase.

**Comment by diguida** [comment]
please test
Amended last commit in order to fix `Configuration/StandardSequences/python/AlCaHarvesting_cff.py` which was issuing a deprecation warning when used in wf `1001.0`. Local tests ok.

**Comment by mmusich** [comment]
+1

**Comment by deguio** [comment]
+1

**Comment by Martin-Grunewald** [comment]
+1

**Comment by ggovi** [comment]
+1

**Comment by ianna** [comment]
+1

**Comment by diguida** [comment]
it does not merge anymore after #13271 is merged (touches https://github.com/cms-sw/cmssw/pull/13271/files#diff-3cbb7f62c930cbb6c3816f38959e9b32). Let's fix conflicts and rebase

**Comment by diguida** [comment]
please test

**Comment by diguida** [comment]
re-running the matrix locally to see if there are leftover warnings :-)

**Comment by diguida** [comment]
@davidlange6 @franzoni @mmusich in order to avoid fixing git merge conflicts, and since all L2 have already signed the changes in this PR, can I suggest it is merged as soon as the tests are ok?

BTW, in order to pick the latest changes by #13271 I had to rebase on top of the current tip of remote `CMSSW_8_0_X` branch, namely https://github.com/cms-sw/cmssw/commit/6a40664d537b9cc30bfc135e9871971c3b3016c8.

**Comment by mmusich** [comment]
please test

**Comment by deguio** [comment]
+1

**Comment by diguida** [comment]
The rebase was needed in order to fix a conflict with the updates by #12651:
[`DQM/Integration/python/clients/hcal_dqm_sourceclient-live_cfg.py`](https://github.com/cms-sw/cmssw/pull/12651/files#diff-cf21d8a06363f8a0dd8cfd6705fb5c83)
[`DQM/Integration/python/clients/hcalcalib_dqm_sourceclient-live_cfg.py`](https://github.com/cms-sw/cmssw/pull/12651/files#diff-ed6ca49352fef94a875e3d78c6dde898). I had to rebase on top of the current tip of remote CMSSW_8_0_X branch, namely https://github.com/cms-sw/cmssw/commit/c4f92f565a58057ab2a1fa83f6e111baf783d8c9

Let me add @vkhristenko here. In the HCAL DQM client applications I changed a bit the logic:
* each application loads [`DQM.Integration.config.FrontierCondition_GT_cfi`](https://github.com/cms-sw/cmssw/pull/13170/files#diff-a80f7fe479a7d3a355e976404dc80bd7): this module loads, via [`Configuration.StandardSequences.FrontierConditions_GlobalTag_cff](https://github.com/cms-sw/cmssw/pull/13170/files#diff-25b5b144a65bd60bcb3e2159dd67a22a) [`CondCore.ESSources.CondDBESSource_cfi`](https://github.com/cms-sw/cmssw/pull/13170/files#diff-efad024d56e42bd52d1ad42d36214da4) where the connection string is centrally set up according to which frontier access is supposed to be used; the DQM module then sets the Global Tag to be used by the client applications.
* in case you want to run offline locally the client for testing purposes, you do not need to load the condition access for offline (according to what stated above), but you only need to specify a suitable GlobalTag. You were choosing the GT to be the HLT GT used for pp running in 2015 with 74X, but this is not supposed to work in future releases. What I have put is a pointer to the Global Tag used for the HLT step by RelVal tests, identified with the `run2_hlt` key in [`Configuration.AlCa.autoCond`](https://github.com/cms-sw/cmssw/blob/CMSSW_8_0_X/Configuration/AlCa/python/autoCond.py). This is done via the indirection string `auto:run2_hlt` in the `gtCustomise` function, as in [`Configuration.AlCa.GlobalTag`](https://github.com/cms-sw/cmssw/blob/CMSSW_8_0_X/Configuration/AlCa/python/Global Tag.py) (check the function there for its usage).

I have also a question concerning the condition customisation you enable via `useMap`, as in https://github.com/cms-sw/cmssw/pull/12651/files#diff-cf21d8a06363f8a0dd8cfd6705fb5c83R116, i.e. the consuming of the HCAL electronics map in your client application. The tag you want to use is actually present in [all data processing HLT/express/prompt GT](https://cms-conddb.cern.ch/cmsDbBrowser/search/Prod/HcalElectronicsMap_v7.05_hlt) and also the page snapshot.
screen shot 2016-02-16 at 14 49 10
Can you please clarify the use-case here? Are you trying to force the unpacking of `uTCA` channels by reading the associated electronics map? Does this require a special tag/payload?

Please be aware that in the GT we have a special HCAL electronics map enabling the readout of all channels, see [minutes of original discussion at AlCaDB](https://indico.cern.ch/event/368332/contribution/1/note/), accessible from GT using label `full`: if so, you should point to the correct unpacker and digi collections. Hence, I suppose you do not need to customise the conditions, but only the client.

**Comment by vkhristenko** [comment]
Hi Salvatore, @diguida 

Let me separate:

1) Conditions. That is fine. The connection string is just set up by importing some central module in my app.  Just like you said..... it might be old - this PR is there since December and I've put what worked in Online at that point........

For offline, I've put that for the reference and put a switch. an idea is that if new people come - that is obvious what to do online/offline locally, etc.. it's just there for the reference. GT for 74 is because this development goes back to pp running with 74 in 2015. 

Totally fine with me - I will figure out the details.

2) So, emap. In online I never did so far. 
The point here is just to have a switch and document for the future.(with such a switch I can also put a txt file and use that - not from GT, which actually happens quite a lot... locally)  
Bottom line, I use that only locally - not in the offical production...

VK

**Comment by diguida** [comment]
Unit test failures are unrelated to this PR:

* `runtestSimCalorimetryHGCalSimProducers`: the failure reported below is the same as in the [`CMSSW_8_0_X_2016-02-15-2300` IB unit test](https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/slc6_amd64_gcc493/CMSSW_8_0_X_2016-02-15-2300/unitTestLogs/SimCalorimetry/HGCalSimProducers)

````
----- Begin Fatal Exception 16-Feb-2016 15:37:45 CET-----------------------
An exception of category 'Filling of conditions failed' occurred while
   [0] Processing run: 1
   [1] Running path 'digitisation_step'
   [2] Calling beginRun for module MixingModule/'mix'
   [3] Using EventSetup component HcalHardcodeCalibrations/'es_hardcode' to make data HcalMCParams/'' in record HcalMCParamsRcd
Exception Message:
 no valid filling possible for Conditions of type HcalMCParams for DetId (0x43400448) (HB -1,72,4)
----- End Fatal Exception -------------------------------------------------
terminate called after throwing an instance of 'cms::Exception'
  what():  An exception of category 'Assert' occurred.
Exception Message:
Geant4 is still alive, master thread state must be set to EndRun before Destruct

/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_0_X_2016-02-15-2300/src/SimCalorimetry/HGCalSimProducers/test/runtests.sh: line 5:  5213 Aborted                 cmsRun ${LOCAL_TEST_DIR}/testHGCalDigi_cfg.py
Failure using testHGCalDigi_cfg.py: status 134
status = 34304

---> test runtestSimCalorimetryHGCalSimProducers had ERRORS
````

* `runtestRecoLocalCaloHGCalRecProducers`: the failure reported below is the same as in the [`CMSSW_8_0_X_2016-02-15-2300` IB unit test](https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/slc6_amd64_gcc493/CMSSW_8_0_X_2016-02-15-2300/unitTestLogs/RecoLocalCalo/HGCalRecProducers)

````
----- Begin Fatal Exception 16-Feb-2016 15:38:00 CET-----------------------
An exception of category 'Filling of conditions failed' occurred while
   [0] Processing run: 1
   [1] Running path 'digitisation_step'
   [2] Calling beginRun for module MixingModule/'mix'
   [3] Using EventSetup component HcalHardcodeCalibrations/'es_hardcode' to make data HcalMCParams/'' in record HcalMCParamsRcd
Exception Message:
 no valid filling possible for Conditions of type HcalMCParams for DetId (0x43400448) (HB -1,72,4)
----- End Fatal Exception -------------------------------------------------
terminate called after throwing an instance of 'cms::Exception'
  what():  An exception of category 'Assert' occurred.
Exception Message:
Geant4 is still alive, master thread state must be set to EndRun before Destruct

/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_0_X_2016-02-15-2300/src/RecoLocalCalo/HGCalRecProducers/test/runtests.sh: line 5:  6984 Aborted                 cmsRun ${LOCAL_TEST_DIR}/testHGCalRecoLocal_cfg.py
Failure using testHGCalRecoLocal_cfg.py: status 134
status = 34304

---> test runtestRecoLocalCaloHGCalRecProducers had ERRORS
````

* `testRecoMETMETProducers`: the failure reported below is not the same as in the [`CMSSW_8_0_X_2016-02-15-2300` IB unit test](https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/slc6_amd64_gcc493/CMSSW_8_0_X_2016-02-15-2300/unitTestLogs/RecoMET/METProducers): here it is a problem with file access (unrelated to conditions), in the IB it is an issue with missing collections.

````
----- Begin Fatal Exception 16-Feb-2016 15:46:40 CET-----------------------
An exception of category 'FileOpenError' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing input source of type PoolSource
   [2] Calling RootFileSequenceBase::initTheFile()
   [3] Calling StorageFactory::open()
   [4] Calling File::sysopen()
Exception Message:
Failed to open the file 'ttbarForMetTests.root'
   Additional Info:
      [a] Input file file:ttbarForMetTests.root could not be opened.
      [b] open() failed with system error 'No such file or directory' (error code 2)
----- End Fatal Exception -------------------------------------------------
Failure using recoMET_genMet_cfg.py: status 84
status = 21504

---> test testRecoMETMETProducers had ERRORS
````

Let's have a look to the `CMSSW_8_0_X_2016-02-16-1100` IB...

**Comment by diguida** [comment]
I can see the same pattern in [the unit tests for the `CMSSW_8_0_X_2016-02-16-1100` IB](https://cmssdt.cern.ch/SDT/cgi-bin/showBuildLogs.py/slc6_amd64_gcc530/www/tue/8.0-tue-11/CMSSW_8_0_X_2016-02-16-1100)

**Comment by diguida** [comment]
In `CMSSW_8_0_X_2016-02-16-2300` IB, at least for gcc530, I see that `runtestSimCalorimetryHGCalSimProducers` and `runtestRecoLocalCaloHGCalRecProducers` are now ok, but `testRecoMETMETProducers` is still failing.

**Comment by diguida** [comment]
please test

**Comment by mmusich** [comment]
+1

**Comment by Martin-Grunewald** [comment]
+1

**Comment by civanch** [comment]
+1

**Comment by ggovi** [comment]
+1

**Comment by diguida** [comment]
@davidlange6 this PR removes all the python deprecation warnings for Condition DB access from the standard sequences (checked in the short matrix, the full one is running locally). Given that it is fully signed and was reported at the ORP meeting, can I suggest it is merged for 8.0.0?

**Comment by davidlange6** [comment]
@diguida - sorry to come very late ... Its now apparent that the plan here is to move the standard conditions cff out of Configuration/StandardSequences. This is quite a change compared to the rest of the software stack. I don't support it. Could you instead use a trivial entry point in that package to make it easier for people to find?

**Comment by diguida** [comment]
@davidlange6 thanks for your comment.
What I moved out of `Configuration/StandardSequences` is the initialisation of the `ESSource` accessing conditions from DB, namely `PoolDBESSource`, for online and offline production configuration: this is done by `CondDBESSource_cfi`. The rationale of this is to put the `PoolDBESSource` module's `cfi` file which specifies the module's parameters and their default values where it should actually be, i.e. in the python directory of the package which contains the module, `CondCore/ESSources`. In fact, the old `FrontierConditions_GlobalTag_cfi` is the only `cfi` in that package which should contain configuration fragments `cff` for all the production sequences.
Also, in order to allow for testing condition access from several DB sources, I left the old `PoolDBESSource_cfi`, so that users can customise the access without touching the production configuration.

What I can do is the following:
* add a new `cff`, namely `Configuration/StandardSequences/python/CondDBESSource_cff.py` importing only CondCore.ESSources.CondDBESSource_cfi: this way, you have a standard configuration fragment as entry point for condition access, which could be eventually customised if *really* needed:

````
from CondCore.ESSources.CondDBESSource_cfi import GlobalTag
````

* leave the new `Configuration/StandardSequences/python/AdditionalConditions_cff.py` as it is, i.e. import of the ancillary modules adding more products to the `EventSetup` based on the DB conditions;
* change `Configuration/StandardSequences/python/FrontierConditions_GlobalTag_cff.py` to load `Configuration.StandardSequences.CondDBESSource_cff`:

````
from Configuration.StandardSequences.FrontierConditions_GlobalTag_cff import *
from Configuration.StandardSequences.AdditionalConditions_cff import *
````

* propagate this to the client code.

What do you think?

**Comment by diguida** [comment]
BTW, a fix is needed anyhow. As @deguio pointed out, there is [a GT customisation in the ECAL client pointing to a CondDBv1 account](https://github.com/diguida/cmssw/blob/conddbV2-remove-V1-deps-6_rebased/DQM/Integration/python/clients/ecal_dqm_sourceclient-live_cfg.py#L125-L134).

**Comment by deguio** [comment]
ciao salvatore,
do you mind updating this PR to fix the ecal problem as well? at least to have something working for next week. we will iterate with the ecal community for a permanent solution.
thanks,
F.

**Comment by diguida** [comment]
@deguio yes sure. I am looking at other offenders, as discussed privately :-)

**Comment by diguida** [comment]
@deguio here you are [the client applications to be updated](https://github.com/search?utf8=✓&q=FrontierOnProd+repo%3Acms-sw%2Fcmssw+path%3ADQM%2FIntegration%2Fpython%2Fclients&type=Code&ref=advsearch&l=&l=)

**Comment by vkhristenko** [comment]
@diguida , hcaltiming application mentioned just above is not being used. Why it exists - no idea (answer from HCAL DQM)

**Comment by diguida** [comment]
@vkhristenko thanks for your input. Do you plan to resurrect it, or can it be dropped completely?

**Comment by vkhristenko** [comment]
@diguida , at this point I can say we git rm on that guy - I don't know what is typically done for these kind of things

**Comment by diguida** [comment]
@vkhristenko deal.
BTW, I owe you a reply to your clarification before (thanks): my web client crashed while typing...

**Comment by davidlange6** [comment]
@diguida - I think I agree most with: 

add a new cff, namely Configuration/StandardSequences/python/CondDBESSource_cff.py

It could even be Conditions_cff.py to be more human friendly... 

then this cff is the standard entry point for consumers.

**Comment by diguida** [comment]
@davidlange6 give a look to https://github.com/diguida/cmssw/commit/3391db0aa0a1905ef9ddc13ea18d2ae4e2a8b06e

**Comment by diguida** [comment]
@deguio give a look to https://github.com/diguida/cmssw/commit/d1f91353c6b0112f59a182a0a164a66080687cfe

**Comment by deguio** [comment]
please test

**Comment by ianna** [comment]
+1

**Comment by deguio** [comment]
+1

**Comment by ggovi** [comment]
+1

**Comment by diguida** [comment]
@davidlange6 is it ok now?

**Comment by diguida** [comment]
I was able to put the branch in good shape for both 80X and 81X with a smart combination of `git rebase --onto` and `git push -f`. Commands are self-explaining:

````
$ git remote add cmsdqm https://github.com/cmsdqm/cmssw.git
$ git fetch cmsdqm
$ git checkout -b consumes80x cmsdqm/consumes80x
$ git checkout -b conddbV2-remove-V1-deps-6_rebased my-cmssw/conddbV2-remove-V1-deps-6_rebased
$ git push my-cmssw HEAD:conddbV2-remove-V1-deps-6_rebased_old
$ git checkout -b consumes80x_and_conddbV2-remove-V1-deps-6_rebased my-cmssw/conddbV2-remove-V1-deps-6_rebased
$ git rebase -i --onto consumes80x CMSSW_8_0_X consumes80x_and_conddbV2-remove-V1-deps-6_rebased
#I had to fix some conflicts here due to some changes in the online DQM applications being cherry-picked by commit https://github.com/cmsdqm/cmssw/commit/0f44709b84b09f394f8b790ab671f111d6a42122
$ git push my-cmssw consumes80x_and_conddbV2-remove-V1-deps-6_rebased
#this creates a new remote branch in my repo
$ git branch -u my-cmssw/consumes80x_and_conddbV2-remove-V1-deps-6_rebased
$ git push my-cmssw +HEAD:conddbV2-remove-V1-deps-6_rebased
#this forces the update to the tip of the rebased branch
````

In this way, this PR in 80X does not show merge conflicts, as the branch has been rebased on top of the one proposed in #13399 thus creating a consistent set of changes.

**Comment by diguida** [comment]
please test

**Comment by diguida** [comment]
please test
Updated to latest Express GT

**Comment by ianna** [comment]
+1

**Comment by mulhearn** [comment]
+1

**Comment by diguida** [comment]
to all other L2s: ping :-)
@davidlange6 I assume you want to merge #13370 in `CMSSW_8_1_X` first

**Comment by mmusich** [comment]
+1

**Comment by perrotta** [comment]
+1

**Comment by ggovi** [comment]
+1

**Comment by fabozzi** [comment]
+1

**Comment by deguio** [comment]
+1

**Comment by diguida** [comment]
please test

**Comment by hengne** [comment]
@franzoni @boudoul  Hi Giovanni and Gaelle, Do you know which part of codes in this PR should PdmV take care? Must be something I didn't touched before.. Thanks!  

best, Hengne

**Comment by diguida** [comment]
@hengne the package requiring PdmV signature is `DPGAnalysis/Skims`, in particular changes in [`DPGAnalysis/Skims/interface/PhysDecl.h`](https://github.com/cms-sw/cmssw/pull/13170/files#diff-b93d104e6ed5f9c39be56a886a6cccc7 and [`DPGAnalysis/Skims/src/PhysDecl.cc`](https://github.com/cms-sw/cmssw/pull/13170/files#diff-cb53b7a772a54d40b26012c51eb285d0). These changes were already proposed in #13399 and merged in the 81X counterpart #13400.
HTH

**Comment by Martin-Grunewald** [comment]
+1

**Comment by hengne** [comment]
+1

**Comment by ianna** [comment]
+1

**Comment by civanch** [comment]
+1

**Comment by deguio** [comment]
+1

**Comment by mmusich** [comment]
+1

**Comment by mulhearn** [comment]
+1

**Comment by ggovi** [comment]
+1

**Comment by dmitrijus** [comment]
Sorry for resurrecting this PR.
Why isn't this merged? Let's make a decision: either to merge it or to close this PR.

**Comment by fwyzard** [comment]
DQM signature is still pending...

**Comment by dmitrijus** [comment]
+1

**Comment by dmitrijus** [comment]
For some unknown reason it was dropped from our p5 tag collector page, so that is why there was no follow-up. I will ping @diguida .

**Comment by diguida** [comment]
@dmitrijus sorry for belated reply.
This PR was never deployed in online DQM: we discussed within DQM and we decided not to deploy it standalone, but wait for central validation, as it completely re-engineers the configuration for condition access in CMSSW.
OTOH, the 81X PR was integrated, the newer pre-releases are running fine with it. As a consequence, we have now two options:

1. we back-port the 81X PR in 80X: this implies closing this PR and making a new one on top of the latest commits in the 80X branch.
2. we decide not to implement this feature in 80X: the PR is closed, and the deployment of 81X for production will feature the revision of the condition access.

Hope this clarifies

**Comment by davidlange6** [comment]
complete re-engineerings are typically not back ported. I'll just close this.
